app/models/site.rb

```diff
@@ -7,6 +7,9 @@ class Site < ApplicationRecord
   scope :at_organizations, ->(names) do
     joins(landscape: :organizations).where(organizations: {organization_name: Array.wrap(names)})
   end
+  scope :for_location, ->(location_id) do
+    where("ST_Intersects(ST_SetSRID(sites.area, 4326), (SELECT ST_GeomFromGeoJSON(geometry) FROM locations WHERE id = ?))", location_id)
```

Comment:
as we are doing spatial intersection I do recomend a small change in the locations api where we do create a geom column so the intersection could be faster.